include/swift/SIL/SILBasicBlock.h

```diff
@@ -558,17 +558,22 @@ struct PhiOperand {
 
   bool operator!=(PhiOperand other) const { return !(*this == other); }
 
-  BranchInst *getBranch() const {
-    return cast<BranchInst>(predBlock->getTerminator());
+  // TODO: This should eventually be limited to BranchInst.
+  TermInst *getBranch() const {
+    return predBlock->getTerminator();
   }
 
   Operand *getOperand() const {
     return &getBranch()->getAllOperands()[argIndex];
   }
 
+  // TODO: This should always return a valid value once critical edges are
+  // illegal.
   SILPhiArgument *getValue() const {
-    auto *branch = cast<BranchInst>(predBlock->getTerminator());
-    return cast<SILPhiArgument>(branch->getDestBB()->getArgument(argIndex));
+    if (auto *branch = dyn_cast<BranchInst>(predBlock->getTerminator()))
+      return cast<SILPhiArgument>(branch->getDestBB()->getArgument(argIndex));
```

Comment:
Same here, should the `PhiOperand` constructor be updated for `cond_br` in non-ossa?

Reply:
For now, I fixed PhiOperand to be invalid for CondBr. We'll either need need prohibit CondBr arguments or change the representation of this abstraction. Prohibiting CondBr arguments would be far simpler in the long term.